configure.ac

```diff
@@ -233,20 +233,25 @@ FI_PROVIDER_SETUP_PC
 
 # If the user requested to build in direct mode, but
 # we have more than one provider, error.
+PROVIDER_DIRECT=no
+FI_DIRECT_PROVIDER_API_10=
 AS_IF([test x"$enable_direct" != x"no"],
-	[AS_IF([test "$PROVIDERS_COUNT" -eq "1"],
-		[AC_SUBST(PROVIDER_DIRECT, "$enable_direct")]
-		[AC_DEFINE_UNQUOTED([FI_DIRECT_PROVIDER_API_10],
-				    ["prov/$enable_direct/provider_FABRIC_1.0.map"],
-				    [Define location of provider symbol script])]
-		AC_DEFINE([FABRIC_DIRECT_ENABLED], 1,
-		  [Define to 1 if building for fi_direct support]),
-		[AC_MSG_NOTICE([Only one provider can be chosen when using --enable-direct])
-		AC_MSG_ERROR(Cannot continue)])])
-
+      [AS_IF([test "$PROVIDERS_COUNT" -gt 1],
+	     [AC_MSG_WARN([Only one provider can be chosen when using --enable-direct])
+	      AC_MSG_ERROR(Cannot continue)])
+
+       PROVIDER_DIRECT=$enable_direct
+       file="prov/$enable_direct/provider_FABRIC_1.0.map"
+       AS_IF([test ! -r $file],
+	     [AC_MSG_WARN([--enable-direct=$enable_direct specified, but $file does not exist])
+	      AC_MSG_ERROR([Cannot continue])])
+       FI_DIRECT_PROVIDER_API_10=`cat $file`])
+
```

Comment:
You could use `AC_SUBST_FILE` instead, [as suggested in my comment on the other PR](https://github.com/ofiwg/libfabric/pull/1828#issuecomment-193302419).